test/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated.test.cpp

```diff
@@ -0,0 +1,969 @@
+
+        testing::internal::CaptureStdout() ;
+
+        EXPECT_NO_THROW(propagatedModel.print(std::cout, true)) ;
+        EXPECT_NO_THROW(propagatedModel.print(std::cout, false)) ;
+
+        EXPECT_FALSE(testing::internal::GetCapturedStdout().empty()) ;
+
+    }
+
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+TEST (OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagated, getEpochANDgetRevolutionNumberAtEpoch)
+{
+
+    using ostk::core::types::Shared ;
+    using ostk::core::types::Real ;
+    using ostk::core::types::Integer ;
+    using ostk::core::ctnr::Array ;
+    using ostk::core::ctnr::Table ;
+    using ostk::core::fs::Path ;
+    using ostk::core::fs::File ;
+
+    using ostk::math::obj::Vector3d ;
+
+    using ostk::physics::units::Length ;
+    using ostk::physics::units::Angle ;
+    using ostk::physics::units::Derived ;
+    using ostk::physics::time::Scale ;
+    using ostk::physics::time::Instant ;
+    using ostk::physics::time::Duration ;
+    using ostk::physics::time::Interval ;
+    using ostk::physics::time::DateTime ;
+    using ostk::physics::coord::Frame ;
+    using ostk::physics::coord::Position ;
+    using ostk::physics::coord::Velocity ;
+    using ostk::physics::Environment ;
+    using ostk::physics::env::obj::celest::Earth ;
+
+    using ostk::astro::trajectory::Orbit ;
+    using ostk::astro::trajectory::State ;
+
+    using ostk::astro::trajectory::orbit::models::Propagated ;
+
+    {
+
+        // Environment setup
+        const Environment environment = Environment::Default() ;
+
+        const Shared<const Frame> gcrfSPtr = Frame::GCRF() ;
+        
+        // Current state and instant setup
+
+        const Instant startInstant = Instant::DateTime(DateTime(2018, 1, 1, 0, 0, 0), Scale::UTC) ;
+
+        const State state = { startInstant, Position::Meters({ 0.0, 0.0, 0.0 }, gcrfSPtr), Velocity::MetersPerSecond({ 1.0, 0.0, 0.0 }, gcrfSPtr) };
+
+        // Orbital model setup
+        const Derived gravitationalParameter = Earth::Models::EGM2008::GravitationalParameter ;
+        const Length equatorialRadius = Earth::Models::EGM2008::EquatorialRadius ;
+
+        const Propagated propagatedModel = { state, startInstant, gravitationalParameter, equatorialRadius, Propagated::GravitationalPerturbationType::None, Propagated::AtmosphericPerturbationType::None, Propagated::ThirdBodyPerturbationType::None , false } ;
+
+        EXPECT_EQ(propagatedModel.getEpoch(),startInstant) ;
+        EXPECT_TRUE(typeid(propagatedModel.getEpoch())==typeid(startInstant)) ; // @BOSS do I need both of these or does the first one suffice cause if .getEpoch() doesn't return an Instant type it would error right? Or is that not necessarily true
```

Comment:
At what stage is the typing enforced? Is it by the compiler when the code is compiled or is it like since an Instant is defined to be returned by the .getEpoch() function then for sure an Instant will be returned? @Derollez